### PR TITLE
Refactor trending to use generic cache driver

### PR DESCRIPTION
**Pull request by Indemnity83**
This refactors the Trending system to use the generic Laravel Cache facade instead of specifically relying on Redis. 

Might be controversial since this implementation is likely technical slower then the Redis specific implementation; however I think the benefit of being able to configure the cache is a worthwhile tradeoff. 

If there is a strong desire to keep the Redis specific implementation I will extract Trending as an interface and add a config value (or piggy back on the cache driver) to load either the generic implementation or Redis specific implementation at runtime.

### CODE REVIEW

**Comment by peterlupu** [comment]
Also, I think we could scrap the iffy `cacheKey()` and use array cache when running tests in `phpunit.xml`.

**Comment by Indemnity83** [comment]
Good call ... I've implemented that for this PR.

File: tests/Feature/TrendingThreadsTest.php [hunk 1]
```diff
@@ -14,8 +14,9 @@ protected function setUp()
     {
         parent::setUp();
 
-        $this->trending = new Trending();
+	config(['cache.default' => 'array']);
```
Comment (bambamboole) [inline tests/Feature/TrendingThreadsTest.php]: I think this will be set in phpunit.xml

Comment (Indemnity83) [reply tests/Feature/TrendingThreadsTest.php]: Yep... brain fart, I removed the explicit config set from the test.

**Pull request merged**